climada/engine/impact_calc.py

```diff
@@ -411,10 +411,13 @@ def impact_matrix(self, exp_values, cent_idx, impf):
         n_exp_pnt = len(cent_idx)  # implicitly checks in matrix assignement whether
                                    # len(cent_idx) == len(exp_values)
         mdr = self.hazard.get_mdr(cent_idx, impf)
-        fract = self.hazard.get_fraction(cent_idx)
         exp_values_csr = sparse.csr_matrix(  # vector 1 x exp_size
             (exp_values, np.arange(n_exp_pnt), [0, n_exp_pnt]),
             shape=(1, n_exp_pnt))
+        if self.hazard.fraction.nonzero()[0].size == 0:
+            return mdr.multiply(exp_values_csr)
+
+        fract = self.hazard.get_fraction(cent_idx)
```

Comment:
Sure, that works well!

Reply:
It is just that the   `hazard.get_fraction` is not the standard getter for the attribute `hazard.fraction`. It is to get the fraction for a subset of centroids. Maybe this should be slightly updated too, so that `get_fraction()` returns simply `fraction` (or `None`)?

Reply:
👍 Good idea. Should be fairly easy too, since the fraction is hardly ever read. I've been looking for occurrences of fraction-readings throughout the code base and apart from this line here I found them only in tests and in hazards for rewriting or plotting.